### PR TITLE
Rustfix update

### PR DESCRIPTION
**Pull request by ehuss**
This updates to rustfix 0.6.0. One of the key changes here is https://github.com/rust-lang/rustfix/pull/195 which changes rustfix to apply multi-part suggestions. One of the tests needs to updated because there are some overlapping suggestions which rustfix cannot handle. The solution is to only apply the machine-applicable suggestions to avoid the overlapping issue.

This also includes a minor change to compiletest to provide better error messages with rustfix.

### CODE REVIEW

**Comment by Mark-Simulacrum** [comment]
Hm, am I correct that the changes in this PR have no bearing on the rustfix users observe in use? That comes from cargo, and is not being updated here, right?

It seems like our compiletest should use the same version as users will see -- I'm surprised to see only one get updated here. Maybe it makes sense to add a tidy check or otherwise assert the two are the same?

**Comment by ehuss** [comment]
Yea, this is only for running compiletest tests.

The reason this is separate is because #86207 needed to be merged to unblock another PR, and there was some desire to unblock it before beta branched.  I didn't want to make the PR wait for these updates to get reviewed and merged.

I went ahead and added a small check to tidy.  rustfix semver-incompatible changes only happen once every few years, so it's not too common.  The tidy check can be removed if it is a problem.

**Review by Mark-Simulacrum — APPROVED** [review]

**Comment by Mark-Simulacrum** [comment]
Sounds good, thanks for adding the tidy check. @bors r+

**Pull request merged**
